src/list.tsx

```diff
@@ -517,8 +508,12 @@ const RowContent = React.forwardRef<TouchableElement, RowContentProps>((props, r
     }
 
     const renderRowWithControl = (Control: typeof Switch | typeof Checkbox) => {
-        const name = props.switch?.name ?? props.checkbox?.name ?? titleId;
-        return props.onPress ? (
+        const name =
+            ((areSwitchRowContentProps(props) && (props.switch?.name ?? undefined)) || undefined) ??
+            ((areCheckboxRowContentProps(props) && (props.checkbox?.name ?? undefined)) || undefined) ??
+            titleId;
```

Comment:
why is this an improvement over the previous code? I find it quite more difficult to read